### PR TITLE
Issue #34545 PHP Version error do not send a proper code error

### LINKED ISSUE #34545: PHP Version error do not send a proper code error
### Summary (*)

If the PHP version is not the one needed by Magento 2, we only get a message and a return code 200.

### Examples (*)

```php
//bootstrap.php
if (!defined('PHP_VERSION_ID') || PHP_VERSION_ID < 70300) {
    if (PHP_SAPI == 'cli') {
        echo 'Magento supports PHP 7.3.0 or later. ' .
            'Please read https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html';
    } else {
        echo <<<HTML
<div style="font:12px/1.35em arial, helvetica, sans-serif;">
    <p>Magento supports PHP 7.3.0 or later. Please read
    <a target="_blank" href="https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html">
    Magento System Requirements</a>.
</div>
HTML;
    }
    exit(1);
}
```

It slows down the potential tracking of issues when multiple servers are involved. Health tracking tools will not see the error because the response is a 200.

I know that this is an infrastructure related issue but it could happen on misconfigured servers, and Magento could help to debug it faster.

The same happens when the Autoload is missing in index.php
 
```php
try {
    require __DIR__ . '/app/bootstrap.php';
} catch (\Exception $e) {
    echo <<<HTML
<div style="font:12px/1.35em arial, helvetica, sans-serif;">
    <div style="margin:0 0 25px 0; border-bottom:1px solid #ccc;">
        <h3 style="margin:0;font-size:1.7em;font-weight:normal;text-transform:none;text-align:left;color:#2f2f2f;">
        Autoload error</h3>
    </div>
    <p>{$e->getMessage()}</p>
</div>
HTML;
    exit(1);
}
```

### Proposed solution

The else case could return an appropriate HTTP response code such as a 503.

```php
if (!defined('PHP_VERSION_ID') || PHP_VERSION_ID < 70300) {
    if (PHP_SAPI == 'cli') {
        echo 'Magento supports PHP 7.3.0 or later. ' .
            'Please read https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html';
    } else {
  http_response_code(503);
        echo <<<HTML
<div style="font:12px/1.35em arial, helvetica, sans-serif;">
    <p>Magento supports PHP 7.3.0 or later. Please read
    <a target="_blank" href="https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html">
    Magento System Requirements</a>.
</div>
HTML;
    }
    exit(1);
}
```
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users with no workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by artskirk**
### Description (*)

If the PHP version is not the one needed by Magento 2, we only get a message and a return code 200. I have added an appropriate HTTP response code 503.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#34545

### Manual testing scenarios (*)

1. Increment current PHP version identifier to reproduce the issue:
    
Capture-1

2. Open Magento in browser
   
   
Capture-2

**Expected result**
The browser could return an appropriate HTTP response code such as a 503

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/bootstrap.php [hunk 1]
```diff
@@ -19,6 +19,7 @@
         echo 'Magento supports PHP 7.4.0 or later. ' .
             'Please read https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements-tech.html';
     } else {
+        http_response_code(503);
```
Comment (ihor-sviziev) [inline app/bootstrap.php]: Please move this line just before "exit(1);"

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@artskirk, could you please clarify, in case when this issue happening, the response is getting cached in Varnish and after your changes - it's not, right?

**Comment by artskirk** [comment]
> @artskirk,
> 
> 1. could you please clarify, in case when this issue happening, the response is getting cached in Varnish and after your changes - it's not, right? 
> 2. Could you sign the Adobe CLA? 


1.  Yes, it's not cached.
2. CLA is signed.

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Pull request merged**